rfhub2/cli/rfhub_importer.py

```diff
@@ -0,0 +1,216 @@
+        """
+        Deletes all existing collections.
+        """
+        self.client.check_communication_with_app()
+        collections = self.client.get_collections()
+        collections_id = {collection['id'] for collection in collections}
+        for id in collections_id:
+            self.client.delete_collection(id)
+        return collections_id
+
+    def import_libraries(self) -> Tuple[int, int]:
+        """
+        Import libraries to application from paths specified when invoking client.
+        :return: Number of libraries loaded
+        """
+        libraries_paths = self.get_libraries_paths()
+        collections = self.create_collections(libraries_paths)
+        loaded_collections = self.add_collections(collections)
+        return len(loaded_collections), sum(v for dict in loaded_collections for v in dict.values())
+
+    def get_libraries_paths(self) -> Set[Path]:
+        """
+        Traverses all given paths and returns set with paths
+        pointing to libraries to import to app.
+        :return: Set of Paths object pointing to libraries to import
+        """
+        self.client.check_communication_with_app()
+        libraries_paths = set()
+        for path in self.paths:
+            libraries_paths.update(self._traverse_paths(Path(path)))
+        if not self.no_installed_keywords:
+            libdir = Path(robot.libraries.__file__).parent
+            libraries_paths.update(self._traverse_paths(Path(libdir)))
+        return libraries_paths
+
+    def _traverse_paths(self, path: Path) -> Set[Path]:
+        """
+        Traverses through paths and adds libraries to rfhub.
+        Helper function for get_library_paths.
+        """
+        valid_lib_paths = set()
+        if self._is_library_with_init(path):
+            valid_lib_paths.add(path)
+        else:
+            for item in path.iterdir():
+                if item.is_dir():
+                    if self._is_library_with_init(item):
+                        valid_lib_paths.add(item)
+                    else:
+                        valid_lib_paths.update(self._traverse_paths(item))
+                elif item.is_file() and self._is_robot_keyword_file(item) and not self._should_ignore(item):
+                    valid_lib_paths.add(item)
+        return valid_lib_paths
+
+    def create_collections(self, paths: Set[Path]) -> List[Collection]:
+        """
+        Creates list of Collection objects from set of provided paths.
+        :param paths: set of paths
+        :return: list of Collection objects
+        """
+        return [self.create_collection(path) for path in paths]
+        # ToDo try except to handle problem with library creation
+
+    def create_collection(self, path: Path) -> Collection:
+        """
+        Creates Collection object from provided path.
+        :param path: Path
+        :return: Collection object
+        """
+        libdoc = LibraryDocumentation(str(path))
```

Comment:
is conversion to string necessary here?

Reply:
Unfortunately, it was failing with Path object. 
`Building library 'F:\Docs\Desktop\rfhub-new\tests\acceptance\fixtures\SingleClassLib\SingleClassLib.py' failed: AttributeError: 'WindowsPath' object has no attribute 'split'` while it has no problems with passing path as a string.